src/figuresClasses.ts

```diff
@@ -1,19 +1,105 @@
-export interface Figure {
 
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'circle' | 'triangle' | 'rectangle';
+
+export interface Figure {
+  color: Color,
+  shape: Shape,
+  getArea(): number,
 }
 
+const checkEachValueMustBiggerThanZero = (
+  a: number,
+  b = Infinity,
+  c = Infinity,
+): void => {
+  if (Math.min(a, b, c) <= 0) {
+    throw new Error('contains 0 or a negative number');
+  }
+};
+
+const checkCorrectLengthsOfTriangleSides = (
+  a: number,
+  b: number,
+  c: number,
+): void => {
+  checkEachValueMustBiggerThanZero(a, b, c);
+
+  const max = Math.max(a, b, c);
+
+  if (((a + b + c - max) - max) <= 0) {
+    throw new Error('uncorrect lengths of the sides relation in a triangle');
+  }
+};
+
+const round = (value: number): number => {
+  return Math.trunc(value * 100) / 100;
+};
+
 export class Triangle implements Figure {
+  color: Color;
 
+  shape: Shape;
+
+  constructor(
+    color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    checkCorrectLengthsOfTriangleSides(a, b, c);
+    this.color = color;
+    this.shape = 'triangle';
```

Comment:
```suggestion
  shape: Shape;

  constructor(
    public color: Color,
    public a: number,
    public b: number,
    public c: number,
  ) {
    checkCorrectLengthsOfTriangleSides(a, b, c);

    this.shape = 'triangle';
```

Reply:
https://dev.to/satansdeer/typescript-constructor-shorthand-3ibd

Reply:
autor's nickname is "satansdeer" )))))